### PR TITLE
add CRL support for gateways

### PR DESCRIPTION
**Pull request by ramaraochavali**
Part of https://github.com/istio/api/pull/3052 . Implements for Gateways. Will followup with another PR for sidecars

- [ ] Ambient
- [ ] Configuration Infrastructure
- [ ] Docs
- [ ] Dual Stack
- [ ] Installation
- [ ] Networking
- [ ] Performance and Scalability
- [ ] Policies and Telemetry
- [ ] Security
- [ ] Test and Release
- [ ] User Experience
- [ ] Developer Infrastructure

### CODE REVIEW

**Comment by ramaraochavali** [comment]
cc: @kfaseela

**Comment by kfaseela** [comment]
Thanks @ramaraochavali . Will you be adding support for ClientTLSSettings in a separate PR?

**Comment by kfaseela** [comment]
Also would be good to check if we have to add additional validations to : https://github.com/istio/istio/blob/master/pkg/config/validation/validation.go#L579

**Comment by ramaraochavali** [comment]
> Will you be adding support for ClientTLSSettings in a separate PR?

Yes. This more targeted for gateways

File: pilot/pkg/networking/core/v1alpha3/gateway_test.go [hunk 1]
```diff
@@ -134,6 +134,80 @@ func TestBuildGatewayListenerTlsContext(t *testing.T) {
...
 		},
+		{
+			name: "mesh SDS enabled, tls mode ISTIO_MUTUAL, CRL specified",
+			server: &networking.Server{
+				Hosts: []string{"httpbin.example.com"},
+				Port: &networking.Port{
+					Protocol: string(protocol.HTTPS),
+				},
+				Tls: &networking.ServerTLSSettings{
+					Mode:  networking.ServerTLSSettings_ISTIO_MUTUAL,
```
[the remark this answers is not in the document]
Comment (ramaraochavali) [reply pilot/pkg/networking/core/v1alpha3/gateway_test.go:145]: What is the expected behaviour for SIMPLE?

[the remark this answers is not in the document]
Comment (kfaseela) [reply pilot/pkg/networking/core/v1alpha3/gateway_test.go:145]: I think for SIMPLE it makes sense only to check at client side, so only when you implement ClientTLSSettings

[the remark this answers is not in the document]
Comment (ramaraochavali) [reply pilot/pkg/networking/core/v1alpha3/gateway_test.go:145]: added tests

File: pkg/config/validation/validation_test.go [hunk 2]
```diff
@@ -1671,6 +1671,23 @@ func TestValidateTlsOptions(t *testing.T) {
...
 		},
+		{
+			"crl specified for SIMPLE TLS",
+			&networking.ServerTLSSettings{
+				Mode:  networking.ServerTLSSettings_SIMPLE,
+				CaCrl: "crl",
+			},
+			"CRL is not supported with SIMPLE TLS", "",
+		},
+		{
```
Comment (ramaraochavali) [inline pkg/config/validation/validation_test.go:1682]: This currently does not give the validation error as we return https://github.com/istio/istio/blob/28f23712a82ccf163c5ee8332c0e95fa7c62f670/pkg/config/validation/validation.go#L634. But I think we should atleast  add warnings. If folks agree, I can change that in a different PR

**Review by kfaseela — APPROVED** [review]

**Comment by ramaraochavali** [comment]
@howardjohn can you PTAL?

**Review by howardjohn — APPROVED** [review]

**Pull request merged**